src/infra/core/src/compact_service_impl.rs

```diff
@@ -0,0 +1,386 @@
+
+            let commit_result = dataset
+                .commit_add_data(
+                    add_data_params,
+                    Some(OwnedFile::new(
+                        data_slice_batch_info.new_file_path.as_ref().unwrap(),
+                    )),
+                    None,
+                    CommitOpts {
+                        block_ref: &BlockRef::Head,
+                        system_time: Some(Utc::now()),
+                        prev_block_hash: Some(Some(&new_head)),
+                        check_object_refs: false,
+                        is_reset_head: false,
+                    },
+                )
+                .await
+                .int_err()?;
+            new_head = commit_result.new_head;
+
+            // Check is new commited block equals to old one
+            // It is possible when `max-slice-size` is smaller
+            // than size of one data slice
+            if data_slice_batch_info.data_slices_batch.len() == 1 {
+                let new_block = chain.get_block(&new_head).await.int_err()?;
+                if let MetadataEvent::AddData(add_event) = new_block.event {
+                    if let Some(data_slice) = add_event.new_data {
+                        if data_slice_batch_info
+                            .data_slices_batch
+                            .first()
+                            .unwrap()
+                            .to_string()
+                            .contains(&data_slice.physical_hash.as_multibase().to_string())
+                        {
+                            continue;
+                        }
+                    }
+                }
+            }
+            old_data_slices.extend(data_slice_batch_info.data_slices_batch.clone());
+        }
+
+        Ok((old_data_slices, new_head))
+    }
+
+    fn remove_old_files(
+        &self,
+        chain_files_info: &ChainFilesInfo,
+        old_data_slices: &Vec<Url>,
+    ) -> Result<(), CompactError> {
+        for file_url in old_data_slices {
+            fs::remove_file(file_url.to_file_path().unwrap()).int_err()?;
+        }
+
+        let new_head_string = chain_files_info.new_head.as_multibase().to_string();
+        for block_file_url in &chain_files_info.block_file_urls {
+            // Stop cleaning up once we reach block where new chain started
+            if block_file_url.as_str().contains(&new_head_string) {
+                break;
+            }
+            fs::remove_file(block_file_url.to_file_path().unwrap()).int_err()?;
+        }
+        Ok(())
+    }
+
+    fn get_random_operation_name_with_prefix(prefix: &str) -> String {
+        use rand::distributions::Alphanumeric;
+        use rand::Rng;
+
+        let mut name = String::with_capacity(10 + prefix.len());
+        name.push_str(prefix);
+        name.extend(
+            rand::thread_rng()
+                .sample_iter(&Alphanumeric)
+                .take(10)
+                .map(char::from),
+        );
+        name
+    }
+}
```

Comment:
I think we have similar algorithms scattered all over the code base. Maybe they deserve to be somehwere in utility crate?

Reply:
This wasn't a high priority, but I really meant using it everywhere in similar situations. This code is not trivial, and I think we copy paste it for container names, network names, ....

Reply:
Changed in all places